jena-tdb1/src/main/java/org/apache/jena/tdb1/solver/SolverLibTDB.java

```diff
@@ -148,16 +148,13 @@ static QueryIterator graphNames(DatasetGraphTDB ds, Node graphNode, QueryIterato
         List<Abortable> killList = new ArrayList<>();
         Iterator<Tuple<NodeId>> iter1 = ds.getQuadTable().getNodeTupleTable().find(NodeId.NodeIdAny, NodeId.NodeIdAny,
                                                                                    NodeId.NodeIdAny, NodeId.NodeIdAny);
+        iter1 = makeAbortable(iter1, killList, execCxt.getCancelSignal());
```

Comment:
I added abort wrapper here as early as possible (before any filters or distinct operations).
Should make the double check (before and after distinct ) unnecessary.